WordPress/gutenberg
# Fix: Viewport width not accounted for in the Site Editor > Patterns view

Fixes #55324

## What?

This PR fixes an issue where the Site Editor's Pattern View does not take into account the Viewport Width defined in the pattern.

### Before

![before](https://github.com/WordPress/gutenberg/assets/54422211/105d9c20-d0f7-4685-b501-26ed535ea425)

### After 

![after](https://github.com/WordPress/gutenberg/assets/54422211/aed76f4f-e551-4a81-8a05-372816f18299)

## Why?

The `BlockPreview` component that is rendering the pattern receives a `viewportWidth` prop to change the viewport width.
However, in the Site Editor's Pattern preview, the `viewportWidth` property is not passed, so it is displayed with a width of `1200px`, which is the default viewport width value of the `BlockPreview` component.

## How?

I passed the pattern's `viewportWidth` as a property to the `BlockPreview` component.

## Testing Instructions

For Emptytheme, create two patterns with defined viewport widths:

<details><summary>test/emptytheme/patterns/width-500.php</summary>

```html
<?php
/**
 * Title: Viewport Width 500
 * Slug: emptytheme/width-500
 * Categories: banner
 * Viewport Width: 500
 */
?>


<p style="font-size:60px">ABCDEFGHIJKLMNOPQRSTUVWXYZABCDEFGHIJKLMNOPQRSTUVWXYZ</p>

```

</details> 

<details><summary>test/emptytheme/patterns/width-1000.php</summary>

```html
<?php
/**
 * Title: Viewport Width 1000
 * Slug: emptytheme/width-1000
 * Categories: banner
 * Viewport Width: 1000
 */
?>


<p style="font-size:60px">ABCDEFGHIJKLMNOPQRSTUVWXYZABCDEFGHIJKLMNOPQRSTUVWXYZ</p>

```

</details> 






### Testing Instructions for Keyboard


- Activate Emptytheme.
- Go to the Site Editor > Patterns.
- Both patterns should have the correct viewport width (you can see this in the width style applied to the iframe element in the pattern).
- Patterns that do not have a viewport width, such as those created in the Site Editor, should still have a default viewport width of 1200px.
